src/yarp.c

```diff
@@ -14436,6 +14436,7 @@ yp_serialize(yp_parser_t *parser, yp_node_t *node, yp_buffer_t *buffer) {
     yp_buffer_append_u8(buffer, YP_VERSION_MAJOR);
     yp_buffer_append_u8(buffer, YP_VERSION_MINOR);
     yp_buffer_append_u8(buffer, YP_VERSION_PATCH);
+    yp_buffer_append_u8(buffer, YP_SERIALIZE_ONLY_SEMANTICS_FIELDS ? 1 : 0);
```

Comment:
`YP_SERIALIZE_ONLY_SEMANTICS_FIELDS` needs to be defined just for this usage.
I think we should `yp_serialize` or the part writing the header to `serialize.c.erb`, but that seemed out of scope of this PR.